src/components/main/Scenario.js

```diff
@@ -1,89 +1,150 @@
-    //         alert("Your hppd must be a number");
-    //         alert("Your numbeds must be a number");
-    //     }
-    //     if(Number(unit)){
-    //         alert("Your unit can't be a number")
-    //     }
-
-    // }
-
     changeHandler = (event) => {
         let name = event.target.name;
         let val = event.target.value;
-		if(name!=='unit') {
-			if(!(/^\+?[1-9][0-9]*$/.test(val)))
-			{
-				alert("Only numbers(positive integers) can be entered");
-				return;
-			}
-		}
-		this.setState({ [name]: val }, () => 
-    	console.log(this.state));
+        this.setState({[name]: val});
     }
 	
+	calculation = (info,num) =>
+    {
+
+            if (info.bedUnit !== '' && info.census !== '' && info.HPPD !== '')
+            {
+
+                let result = parseInt(info.bedUnit) + parseInt(info.census) + parseInt(info.HPPD);
+                console.log(num + ":" + result)
+
+                if (num <= 0)
+                {
+                    this.setState({ "staffNum": "The Results:" + result });
+                }
+
+                else
+                {
+                    num = result + num;
+                    this.setState({ "staffNum": "The Results:" + num });
+                }
+
+            }
+
+	}
+	
+	setInfo = (name,value) =>
+	{
+		let info=this.state.info;
+		if(name==='unit')
+		{
+			info.unit=value;
+		}else if(name==='HPPD')
+		{
+			info.HPPD=value;
+		}else if(name==='census')
+		{
+			info.census=value;
+		}else if(name==='bedUnit')
+		{
+			info.bedUnit=value;
+		}
+		this.setState({info:info});
+		this.calculation(info,this.state.num);
+	}
 	setInfoStaffNum = (num) =>
 	{
 		debugger;
 		this.setState({"num":num});
-		// this.calculation(this.state.info,num);
+		this.calculation(this.state.info,num);
+	}
+	handleAdd = (e) =>
+	{
+		const name = e.target.name;
+		const value = e.target.value;
+		if(name!=='unit')
+		{
+			if(!(/^\+?[1-9][0-9]*$/.test(value)))
+			{
+				alert("Only numbers(positive integers) can be entered");
+				return;
+			}
+		}
+		
+		
+		this.setInfo(name,value)	
+		
 	}
 	
+	random = (Min,Max) =>{
+		var Range = Max - Min;   
+		var Rand = Math.random();   
+		return (Min + Math.round(Rand * Range));
+	}
+	handleRandom = () =>
+	{
+		const num1 = this.random(8,24);
+		var a=document.getElementById("HPPD");
+		a.value=num1;
+		this.setInfo("HPPD",num1);	
+		
+		const num2 = this.random(1,100);
+		a=document.getElementById("census");
+		a.value=num2;
+	    this.setInfo("census",num2)	;
+		const num3 = this.random(10,1000);
+		a=document.getElementById("bedUnit");
+		a.value=num3;
+		this.setInfo("bedUnit",num3)	;
+    }
+
     render() {
         return (
-
             <div className="App">
 
                     <Card id="scenario">
                         <Card.Header>Fill out the scenario fields based off of the in-class example.</Card.Header>
-						<Card.Body>
-							<Form>
-								<InputGroup size="sm" className="mb-3">
```

Comment:
Is there another place where we are putting the test ID? It looks like it was removed and if the data-testid is removed it will break the input tests.